app/tests/test_collections_model.py

```diff
@@ -1,737 +1,73 @@
-
-        y100 = self.add_single_datetime_item(self.y100, 'y100')
-        y9000 = self.add_single_datetime_item(self.y9000, 'y9000')
-        y200_y8000 = self.add_range_item(self.y200, self.y8000, 'y200_y8000')
-
-        self.assertEqual(
-            self.collection.extent_start_datetime,
-            y100.properties_datetime,
-            "Updating temporal extent (extent_start_datetime) based on mixed "
-            "items failed."
-        )
-
-        self.assertEqual(
-            self.collection.extent_end_datetime,
-            y9000.properties_datetime,
-            "Updating temporal extent (extent_end_datetime) based on mixed "
-            "items failed."
-        )
-
-        y100.properties_datetime = self.y200
-        y100.full_clean()
-        y100.save()
-        y9000.properties_datetime = self.y8000
-        y9000.full_clean()
-        y9000.save()
-        self.collection.refresh_from_db()
-
-        self.assertEqual(
-            self.collection.extent_start_datetime,
-            y100.properties_datetime,
-            "Updating temporal extent (extent_start_datetime) based on mixed "
-            "items failed."
-        )
-
-        self.assertEqual(
-            self.collection.extent_end_datetime,
-            y9000.properties_datetime,
-            "Updating temporal extent (extent_end_datetime) based on mixed "
-            "items failed."
-        )
-
-        y100_y9000 = self.add_range_item(self.y100, self.y9000, 'y100_y9000')
-
-        self.assertEqual(
-            self.collection.extent_start_datetime,
-            y100_y9000.properties_start_datetime,
-            "Updating temporal extent (extent_start_datetime) based on mixed "
-            "items failed."
-        )
-
-        self.assertEqual(
-            self.collection.extent_end_datetime,
-            y100_y9000.properties_end_datetime,
-            "Updating temporal extent (extent_end_datetime) based on mixed "
-            "items failed."
+class CollectionsModelTestCase(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.collection = CollectionFactory().create_sample(db_create=True)
+
+    def test_create_already_existing_collection(self):
+        # try to create already existing collection twice
+        with self.assertRaises(ValidationError, msg="Existing collection could be re-created."):
+            collection = Collection(**self.collection.attributes)
+            collection.full_clean()
+            collection.save()
+
+    def test_create_collection_invalid_name(self):
+        # try to create a collection with invalid collection name
+        with self.assertRaises(ValidationError, msg="Collection with invalid name was accepted."):
+            collection_data = CollectionFactory().create_sample(sample="collection-invalid")
+            collection = Collection(**collection_data.attributes)
+            collection.full_clean()
+            collection.save()
```

Comment:
For consistency I would do it like below, using the `db_create=True`.

Reply:
Done